src/faebryk/libs/util.py

```diff
@@ -957,7 +957,7 @@ def debugging() -> bool:
     return debugpy.is_client_connected()
 
 
-class FuncSet[T, H: Hashable](collections.abc.Set[T]):
+class FuncSet[T, H: Hashable](collections.abc.MutableSet[T]):
```

Comment:
What you are trying to implement is a custom eq function instead of a custom hasher that is removing collisions.
Just add an extra arg eq that checks if two elements (with the same hash) are equivalent and if so dont add it to the set.

Then no need for the discard stuff.

Reply:
I was using `discard` in the DSL front-end to keep track of the params that'd been referenced w/o declaration.
Maybe there's something else fucky with this implementation?